test-packages/integration-tests/test/proxy.spec.ts

```diff
@@ -43,8 +44,98 @@ describe('proxy', () => {
     await closeServer(httpServer);
     await closeServer(proxyServer);
   });
+
+  describe('should have proxy config set', () => {
+    const destination: Destination = {
+      name: 'TESTINATION',
+      url: 'http://example.com',
+      proxyConfiguration: {
+        host: 'proxy.com',
+        protocol: Protocol.HTTP,
+        port: 1234
+      }
+    };
+
+    let requestSpy;
+
+    beforeEach(() => {
+      nock('http://example.com')
+        .get(/.*/)
+        .reply(200, { d: { results: [] } });
+
+      requestSpy = jest.spyOn(axios, 'request');
+    });
+
+    afterEach(() => {
+      requestSpy.mockClear();
+      delete process.env.destinations;
+    });
+
+    it('not for OData requests without proxy configuration', async () => {
+      await TestEntity.requestBuilder()
+        .getAll()
+        .execute({ url: destination.url });
+      expect(requestSpy).not.toHaveBeenCalledWith(objectContainingProxyAgent());
+    });
+
+    it('for OData requests with proxy configuration', async () => {
+      await TestEntity.requestBuilder().getAll().execute(destination);
+      expect(requestSpy).toHaveBeenCalledWith(objectContainingProxyAgent());
+    });
+
+    it('for destinations in env variable', async () => {
+      setTestDestination({ ...destination, name: destination.name });
+      await TestEntity.requestBuilder()
+        .getAll()
+        .execute({ destinationName: destination.name! });
+      expect(requestSpy).toHaveBeenCalledWith(objectContainingProxyAgent());
+    });
+
+    it('for destinations in env variable with proxy env variable', async () => {
+      setTestDestination({ url: destination.url, name: destination.name });
+      process.env.http_proxy = `${destination.proxyConfiguration?.protocol}://${destination.proxyConfiguration?.host}:${destination.proxyConfiguration?.port}`;
```

Comment:
I like the optional chaining, but since the `destination` is a const, why you have the null check here?

Reply:
Because in the destination type the proxyConfiguration is optional.